.github/workflows/cargo-deny.yml

```diff
@@ -0,0 +1,42 @@
+name: cargo deny checks
+
+on:
+  pull_request:
+    paths:
+      - 'Cargo.lock'
+  schedule:
+    - cron: '0 0 * * *'
```

Comment:
For PRs we don't want to have a separate workflow, but rather explicitly call a reusable workflow from the `build_and_test` workflow, reasoning is here: https://github.com/neondatabase/neon/blob/7d4bfcdc4795654100b87a02d398e41373027e44/.github/workflows/build_and_test.yml#L1257-L1265

Reply:
This makes the bit of filtering it to `Cargo.lock` changes a bit more complex, but is still doable. In some places we use `tj-actions/changed-files` ([4 results](https://github.com/search?q=org%3Aneondatabase+%22uses%3A+tj-actions%2Fchanged-files%22&type=code)), in others we use `dorny/paths-filter` ([18 results](https://github.com/search?q=org%3Aneondatabase%20%22uses%3A%20dorny%2Fpaths-filter%22&type=code)). Should we standardize on one action for this?